### PR TITLE
Hard-code the versions of these smoke apps to avoid issue re-using branch version

### PR DESCRIPTION
**Pull request by mcculls**

### CODE REVIEW

**Review by PerfectSlayer — APPROVED** [review]
It might be clearer to not use version number at all and clear it.

It might be worth it to have a look at `test-published-dependencies/build.gradle` later too.

I approve the PR right now to not block the fix further 😉

**Review by richardstartin — APPROVED** [review]

File: dd-smoke-tests/quarkus/build.gradle [hunk 1]
```diff
@@ -42,7 +42,7 @@ tasks.named("compileTestGroovy").configure {
 }
 
 tasks.withType(Test).configureEach {
-  jvmArgs "-Ddatadog.smoketest.quarkus.uberJar.path=$appBuildDir/quarkus-smoketest-$version-runner.jar"
+  jvmArgs "-Ddatadog.smoketest.quarkus.uberJar.path=$appBuildDir/quarkus-smoketest--runner.jar"
```
Comment (mcculls) [inline dd-smoke-tests/quarkus/build.gradle:45]: Note: the Quarkus uber-jar code doesn't collapse the runner name when the version is blank, so we end up with `--` in the name. I couldn't find any way to override this behaviour so have left this double-dash in place.

Comment (PerfectSlayer) [reply dd-smoke-tests/quarkus/build.gradle:45]: You can use `finalName` property to setup its name to whatever you like.

Comment (mcculls) [reply dd-smoke-tests/quarkus/build.gradle:45]: Nope, the quarkus uber-jar task doesn't respect that property or `archiveName` 

We could see if upgrading Quarkus helps, but TBH that's a non-trivial effort just to remove a dash nobody really sees.

Comment (bantonsson) [reply dd-smoke-tests/quarkus/build.gradle:45]: Got to ❤️ the consistent and robust behavior of the `gradle` plugin ecosystem 😉

**Pull request merged**